### PR TITLE
Fix/vertex identifier

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: include/geode/basic/attribute.h [hunk 1]
```diff
@@ -208,9 +184,9 @@ namespace geode
 
         void delete_elements( const std::vector< bool >& to_delete ) override
         {
-            auto nb_removed_elements =
-                detail::compress_vector( to_delete, values_ );
-            resize( values_.size() - nb_removed_elements );
+            std::vector< bool > to_keep( to_delete );
+            to_keep.flip();
```
Comment (BotellaA) [inline include/geode/basic/attribute.h]: Interesting method !

File: src/geode/georepresentation/core/vertex_identifier.cpp [hunk 2]
```diff
@@ -189,6 +199,28 @@ namespace geode
                 } );
         }
 
+        void filter_component_vertices( const uuid& component_id )
+        {
+            for( auto uv_id : Range( nb_unique_vertices() ) )
```
Comment (BotellaA) [inline src/geode/georepresentation/core/vertex_identifier.cpp]: Range{}

File: src/geode/georepresentation/core/vertex_identifier.cpp [hunk 3]
```diff
@@ -189,6 +199,28 @@ namespace geode
...
+                    mesh_component_vertices.size(), true );
+                for( auto i : Range( mesh_component_vertices.size() ) )
+                {
+                    if( mesh_component_vertices[i].component_id.id()
+                        == component_id )
+                    {
+                        to_keep[i] = false;
+                    }
+                }
+                mesh_component_vertices =
```
Comment (BotellaA) [inline src/geode/georepresentation/core/vertex_identifier.cpp]: Maybe call this only if you went on line 216

File: upgrade-guide.md [hunk 4]
```diff
@@ -5,3 +5,33 @@
...
+const geode::Line3D& line = get_a_line();
+vertex_identifier.remove_component( line.id() );
+```
+
+*v1.0.0:*
+
+```
+const geode::Line3D& line = get_a_line();
+vertex_identifier.remove_component( line );
+```
```
Comment (BotellaA) [inline upgrade-guide.md]: Great !

File: src/geode/georepresentation/core/vertex_identifier.cpp (hunk 3 above)
Comment (panquez) [reply src/geode/georepresentation/core/vertex_identifier.cpp]: Good idea. I also add checks in functions extract/delete_vector_elements to escape if nothing to do.

**Pull request merged**